pkg/apis/configuration/v1/kongclusterplugin_types.go

```diff
@@ -35,6 +35,7 @@ import (
 // +kubebuilder:printcolumn:name="Disabled",type=boolean,JSONPath=`.disabled`,description="Indicates if the plugin is disabled",priority=1
 // +kubebuilder:printcolumn:name="Config",type=string,JSONPath=`.config`,description="Configuration of the plugin",priority=1
 // +kubebuilder:printcolumn:name="Programmed",type=string,JSONPath=`.status.conditions[?(@.type=="Programmed")].status`
+// +kubebuilder:validation:XValidation:rule="(has(self.config) || has(self.configFrom)) ? ((!has(self.config) && has(self.configFrom)) || (has(self.config) && !has(self.configFrom))) : true", message="KongClusterPlugin cannot use both Config and ConfigFrom"
```

Comment:
```suggestion
// +kubebuilder:validation:XValidation:rule="(has(self.config) || has(self.configFrom)) ? ((!has(self.config) && has(self.configFrom)) || (has(self.config) && !has(self.configFrom))) : true", message="Using both config and configFrom fields is not allowed."
```

I'd suggest using lowercase names of fields (matching JSON tags) because I believe YAMLs will be the most common way to install plugins and that will make the message more useful for the user. 

In `KongUpstreamPolicy` I also decided to not use the resource kind name in the message as it is already displayed in the error message that `kubectl` gives:
```
The KongPlugin "plugin-one" is invalid: <nil>: Invalid value: "object": KongPlugin cannot use both Config and ConfigFrom
```

Raising this mostly for consistency's sake - happy to discuss which approach is better so we can stick to just one.

Reply:
Good call, let's be consistent.

Reply:
BTW I tried to figure out the `<nil>: Invalid value: "object":` but couldn't make it more user friendly. Any thoughts on this one? Trying to use the `type(self)` in the expression didn't work.

Reply:
Yeah, I also failed to make it disappear/display something more meaningful. 😬